tools/python/influx-migration/influx_migration.py

```diff
@@ -0,0 +1,1115 @@
+        logging.error(str(error))
+        return False
+    finally:
+        client.close()
+    return True
+
+def write_bucket_from_csv(path, host, token, bucket_name, new_bucket_name=None, org=None, skip_verify=False):
+    """
+    Creates a bucket in the destination instance and writes bucket data from a csv file to that
+    created bucket.
+
+    :param path: The path to the backed up data.
+    :type path: class: pathlib.Path
+    :param str host: The host of the destination instance.
+    :param str token: The InfluxDB token for the destination instance.
+    :param str bucket_name: The name of the bucket that was backed up.
+    :param new_bucket_name: Optional, the name of the bucket to create in the destination instance and restore
+        data to.
+    :type new_bucket_name: str or None
+    :param org: The name of the organization in the destination instance to restore to.
+    :type org: str or None
+    :param bool skip_verify: Whether to verify TLS certificates during requests.
+    :returns: None
+    :raises OSError: If reading from the csv file fails.
+    :raises subprocess.CalledProcessError: If writing to the database fails.
+    :raises RuntimeError: If no records were written to the database but no other exception was raised.
+    :raises JSONDecodeError: If the metadata for the bucket could not be loaded.
+    """
+    try:
+        with open(str(path) + f"/metadata_{bucket_name}.json") as metadatafile:
+            metadata = json.load(metadatafile)
+        retention_seconds = metadata["retention_rules"][0]["every_seconds"]
+        shard_group_retention_seconds = metadata["retention_rules"][0]["shard_group_duration_seconds"]
+        org_name = metadata['org_name'] if org is None else org
+        dest_bucket_name = bucket_name
+
+        if new_bucket_name is not None:
+            dest_bucket_name = new_bucket_name
+        logging.info(f"Restoring to bucket {dest_bucket_name}")
+
+        retention_rules = BucketRetentionRules(type='expire', every_seconds=retention_seconds)
+
+        # There is a bug with creating a bucket where if retention is 0 (infinite) and
+        # shard group duration is defined at all then retention will instead be set to 720
+        # hours, risking excluding data points from insertion due to being out of range of
+        # retention policy
+        # Thread: https://community.influxdata.com/t/bucket-created-with-retention-0-shows-720h-retention/30184
+        if retention_seconds > 0:
+            retention_rules.shard_group_duration_seconds = shard_group_retention_seconds
+
+        written_records = 0
+        client = InfluxDBClient(url=host, token=token,
+            verify_ssl=not skip_verify, timeout=MILLISECOND_TIMEOUT, org=org_name)
+        client.buckets_api().create_bucket(bucket_name=dest_bucket_name, org=org_name, retention_rules=retention_rules)
+        write_command = ['influx', 'write', '--host', host, '-t', token, '--bucket', dest_bucket_name,
```

Comment:
This had a hardcoded `--skip-verify` flag that was not caught in review and I have since removed. `--skip-verify` is added below if users have used the `--skip-verify` flag for the script.